src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[10]; // Ініціалізація масиву з початковим розміром
```

Comment:
The initial capacity of the array is hardcoded as 10. According to the checklist, magic numbers should be replaced with constants. Consider defining a constant for the initial capacity.